projects/web-wallet/wallet-connect/0001-design.md

```diff
@@ -0,0 +1,774 @@
+
+> The Web3Wallet SDK allows you to easily connect your Web3 wallet with `dApps`. It provides a simple and intuitive interface for `dApps` to request actions such as signing transactions and interacting with smart contracts on the blockchain. The SDK is designed to work with any blockchain, making it easy to support multiple chains without the need for separate integration code. (https://docs.walletconnect.com/2.0/web3wallet/about)
+
+It's basically a library that provides a single API for the [Sign API](https://docs.walletconnect.com/2.0/api/sign) and the [Auth API](https://docs.walletconnect.com/2.0/api/auth)
+
+#### WalletConnect Cloud
+
+The javascript SDK requires a `projectId` which is supposed to be registered at their WalletConnect cloud, so it can use the WalletConnect infrastructure "for free". 
+
+Note: There is currently no stable open-source implementation of the relay server infrastructure, so this step is really necessary.
+
+<a name="guide_level_explanation"/>
+
+## Guide-level explanation
+
+To go along with this design document, a proof-of-concept has been developed which can be accessed [here](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/245)
+
+The main design decision here is to keep all the business-logic inside the mobile app (or the wallet-desktop), only exposing a JSON-RPC API to the `dApps`, being strict on what methods we want to make available
+
+<a name="guide_level_explanation__ui"/>
+
+### UI
+
+There are two parts of the UI when using wallet-connect: The `dApp` UI, which should basically be a QRCode with the information to connect, as described in the "How does it work?" section of this document and that is already provided with the `Web3Modal` library, described in the SDK section of this document.
+
+The other part would be the wallet, which can be either our already existing mobile-wallet or the desktop-wallet, or both.
+
+On the proof-of-concept, I added the WalletConnect to the wallet-mobile project, here is an example of scanning, pairing and answering a JSON-RPC request:
+
+
+<video src="https://user-images.githubusercontent.com/3586068/225664948-fd45d6ff-6e1a-4585-99e0-6ea2b914ee54.mov"></video>
+
+
+<a name="guide_level_explanation__api"/>
+
+### API
+
+For the first implementation of the WebWallet, I suggest we expose the following JSON-RPC APIs
+
+
+#### 1. Prove ownership of an address
+
+`dApps` might ask the wallets to sign an arbitrary message using an address, this method will receive a string and return a string with the signed message encoded in `base64`
+
+This method should be similar to the Ethereum's `personal_sign` which is part of the Ethereum JSON-RPC interface.
+
+The idea is to allow `dApps` to request an arbitrary data to be signed with a specific address private key, to be able to prove the ownership of that address.
+
+
+#### 2. Sign a transaction
+
+The `dApp` should be able to send a transaction to have its inputs validated and signed from the wallets
```

Comment:
We should highlight that:

1. The wallet receives a raw transaction and should parse it itself.
2. The wallet should gather token information from a trusted backend/full node.
3. The wallet should gather the inputs from a trusted backend/full node.

Reply:
Done in ad6901fc8ae53115be5fe4ea2b1a5485539761bf